### PR TITLE
Refactor transaction

### PR DESCRIPTION
**Pull request by dranikpg**
Me and Adi are now actively adding changes to the Transaction class, so I decided to clean it up a little:

- The header was really chaotic, I ordered the parts logically, so you can get an overview now
- I added some missing comments and updated deprecated ones
- ~~I replaced all the manual index accesses `SidToId` with a `PerShardStorage` that has an operator[](ShardId)~~
- I improved some wicked parts like directly accessing LockCnt's inner array without any explanations

### CODE REVIEW

**Comment by romange** [comment]
I almost LGTMed, but then I saw changes in transaction.h and some other places.

this won't pass as is. I suggest to do the following:
1. Do all the renames without changing the functionality at all. 
2. Do all the reorders/internal changes in transaction.h without changing all the functionality at all.
3. Do all the functionality changes. I consider removing SidToId as changing functionality.

File: src/server/transaction.cc [hunk 1]
```diff
@@ -338,7 +337,7 @@ bool Transaction::RunInShard(EngineShard* shard) {
     DCHECK(!awaked_prerun);  // we should not have blocking transaction inside multi block.
 
     sd.local_mask |= KEYLOCK_ACQUIRED;
-    shard->db_slice().Acquire(mode, GetLockArgs(idx));
+    shard->db_slice().Acquire(mode, GetLockArgs(shard->shard_id()));
```
Comment (romange) [inline src/server/transaction.cc:340]: I need to see why it's correct. Specifically, we keep all the data in the 0 index for single-shard transactions.
I am guessing you move this logic to GetLockArgs but right now I do not see how I can review all the reorderings and functionality changes AND keep my sanity.

**Review by romange — COMMENTED** [review]
Transaction 

my-precious-wool-my-precious

**Comment by dranikpg** [comment]
Divided it into 4 commits. Decided not to include the `SidToId` changes. We can leave it without it for now 🙂

**Comment by romange** [comment]
To summarize - the only functional change here is inside MultiData?

**Comment by dranikpg** [comment]
> To summarize - the only functional change here is inside MultiData?

Yes

**Pull request merged**